test/jest-light-runner/src/index.js

```diff
@@ -0,0 +1,56 @@
+import { Piscina } from "piscina";
+import supportsColor from "supports-color";
+import { MessageChannel } from "worker_threads";
+import os from "os";
+
+/** @typedef {import("@jest/test-result").Test} Test */
+
+const piscina = new Piscina({
```

Comment:
Did you test if `jest-worker` (default is process-based) can improve test performance? Back in https://github.com/babel/babel/pull/13423 I tried both `jest-worker` and `piscina` for the building task and `jest-worker` wins. I don't know if that is true for test, too. 

The `jest-worker` has `enableWorkerThreads` option but it is not exposed to Jest config, I guess process-based runner could be still faster than worker threads.

Reply:
`jest-worker` is a bit faster (31s vs 33s), but it:
- doesn't support ESM, so I must write the test runner as CJS
- doesn't support signaling to the main thread when a task starts, so I cannot show "real time" which tests are being run (this is _slightly_ annoying when the output is stuck on long-running tests like `babel-cli`, because it looks like it's frozen).
- for some reason jest doesn't exit gracefully, even if I call `.end()` to close `jest-worker`'s pool :thinking: